src/UIControls/MenuPaging.ts

```diff
@@ -0,0 +1,145 @@
+import { basePath, sgWorld } from "../Axiom";
+import { Button } from "../Button";
+import { Quaternion } from "../math/quaternion";
+import { Vector } from "../math/vector";
+import { radsToDegs } from "../Mathematics";
+import { Menu } from "../Menu";
+import { GetDeviceType, DeviceType, ProgramManager } from "../ProgramManager";
+
+
+export class MenuPaging extends Menu {
+
+  // override so we have a fixed size
+  cols: number = 3;
+  rows: number = 3;
+  buttonSize: number = 0.1;
+
+  // for paging
+  private pageNumber: number = 0;
+  private totalPages: number = 0;
+  private btnPL!: Button;
+  private btnPR!: Button;
+
+  constructor(public width: number, public height: number, public anchor: Vector<3>, public orientation: Quaternion, public anchorPosition: [number, number], public topAligned: boolean, public leftAligned: boolean, public horizontal: boolean) {
+    super(width, height, anchor, orientation, anchorPosition, topAligned, leftAligned, horizontal);
+
+    // recalculate the anchor so that it is centred. @Ruben better way to do this in the base class as an option
+    // @ruben better way to do this?
+    const newX = anchor.data[0] - ((this.cols * this.buttonSize) / 2);
+    anchor.data[0] = newX
```

Comment:
I think what you want to do is use an anchorPosition of [0.5, y] to make the anchor coordinates the center of the menu?